lib/DXIL/DxilUtil.cpp

```diff
@@ -479,20 +479,6 @@ bool IsHLSLObjectType(llvm::Type *Ty) {
   return false;
 }
 
-bool IsHLSLMatrixType(Type *Ty) {
-  if (StructType *ST = dyn_cast<StructType>(Ty)) {
-    Type *EltTy = ST->getElementType(0);
-    if (!ST->getName().startswith("class.matrix"))
-      return false;
-
-    bool isVecArray =
-        EltTy->isArrayTy() && EltTy->getArrayElementType()->isVectorTy();
-
-    return isVecArray && EltTy->getArrayNumElements() <= 4;
-  }
```

Comment:
This also returns true for arrays of vectors while HLMatrixType::isa checks only the struct type and name. Isn't that a problem?

Reply:
What it's doing is actually checking that the structure is composed of an array of vectors, but this is redundant with the name check because all matrices are. At best it could have been an assert.

Reply:
And what custom user structs that fit this?

struct MyMatrix {
  float4 rows[4];
};

Reply:
That won't pass the name test, which is why the array-of-vector test is redundant.

Reply:
Oh I see :)